app/src/org/commcare/android/view/c3/AxisConfiguration.java

```diff
@@ -0,0 +1,177 @@
+package org.commcare.android.view.c3;
+
+import org.commcare.android.util.InvalidStateException;
+import org.commcare.suite.model.graph.Graph;
+import org.commcare.suite.model.graph.GraphData;
+import org.commcare.suite.model.graph.SeriesData;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+
+/**
+ * Axis-related configuration for C3.
+ *
+ * Created by jschweers on 11/16/2015.
+ */
+public class AxisConfiguration extends Configuration {
+    public AxisConfiguration(GraphData data) throws JSONException, InvalidStateException {
+        super(data);
+
+        JSONObject x = new JSONObject();
+        JSONObject y = new JSONObject();
+        JSONObject y2 = new JSONObject();
+
+        if (Boolean.valueOf(mData.getConfiguration("show-axes", "true")).equals(Boolean.FALSE)) {
+            JSONObject show = new JSONObject("{ show: false }");
+            x = show;
+            y = show;
+            y2 = show;
+        } else {
+            // Undo C3's automatic axis padding
+            JSONObject padding = new JSONObject("{top: 0, right: 0, bottom: 0, left: 0}");

```

Comment:
I would lift all this into a helper function called `addAxisToGraph(JSONObject x, JSONObject y, JSONObject y2)` or something